CourseSearchAPIHttpTrigger/__init__.py

```diff
@@ -132,37 +136,53 @@ def main(req: func.HttpRequest) -> func.HttpResponse:
         facets = response_with_facets.json()
 
         counts = {}
-        # Step 7 - handle facets to build correct
+        # Step 8 - handle facets to build correct
         # limit and offset for next query
         query_params["limit"], query_params["offset"], counts["institutions"], counts[
             "courses"
-        ] = helper.get_offset_and_limit(
+        ], institution_course_counts = helper.get_offset_and_limit(
             facets["@search.facets"]["course/institution/sort_pub_ukprn_name"],
             int(limit),
             int(offset),
         )
 
-        # Step 8 - Build course query
-        search_query = query.build_course_search_query(
-            course, institution, institutions, postcode_object, query_params
-        )
+        # Step 9 Get courses by institution based on query paramaeters
+        # Azure search can only return a maximum of 1000 docs in one call, so page through results
+        if query_params["limit"] > 1000:

```

Comment:
Consideration for future refactor perhaps. This area of code may be simplified a bit if there weren't two calls for getting coureses - one called get_results and one called get_courese. Better if the caller doesn't need to worry about how many courses and handling of more than 1K or not is abstracted behind the function get_coureses.

Reply:
I will leave a TODO comment for now, but yes I totally agree and would have liked to have done this if time wasn't such a constraint